packages/core/src/geometry/GeometrySystem.js

```diff
@@ -588,21 +588,32 @@ export default class GeometrySystem extends System
     {
         const { gl } = this;
         const geometry = this._activeGeometry;
+        const uints_for_indices = gl.getExtension("OES_element_index_uint");
 
         // TODO.. this should not change so maybe cache the function?
 
         if (geometry.indexBuffer)
         {
-            if (geometry.instanced)
```

Comment:
should we be checking for the geometry.indexBuffer.data array type?

also instead of having 4 different checks we could use:
```
const byteSize = geometry.indexBuffer.BYTES_PER_ELEMENT;
const arrayType = byteSize === 2 ?  gl.UNSIGNED_SHORT :  gl.UNSIGNED_INT :

...
gl.drawElementsInstanced(type, size || geometry.indexBuffer.data.length, arrayType, (start || 0) * byteSize, instanceCount || 1);
...
```